### ISSUE TITLE
no visible global function definition for ‘mutate’ object_usage_linter

### ISSUE THREAD

**Issue by USMortality** [r-lib/lintr#2050]
![Screenshot 2023-08-07 at 3 09 32 PM](https://github.com/r-lib/lintr/assets/94856880/b48841f9-31da-4fc3-acf1-8fd5bc940be7)
Why does lintr not recognize the global definition of mutate? I don't want to disable the linter, what's the fix?

**Comment by MichaelChirico** [comment]
Could you give more context about your setup?

Is this a sequence of scripts `a.r` and `b.r`, or is this part of a package?

{lintr} will not execute `source()`, so I'm not sure how it would find your definition.

If this is a package, `#' @importFrom dplyr mutate` should work.

**Comment by USMortality** [comment]
Unfortunately, adding `#' @importFrom dplyr mutate` doesn't seem to work either... I'm using Visual Studio, is this an issue with the VS plugin itself, maybe?
![Screenshot 2023-08-07 at 3 41 22 PM](https://github.com/r-lib/lintr/assets/94856880/43b0fa45-0f14-4769-8d84-744b4e83243b)

**Comment by MichaelChirico** [comment]
That is only expected to work from a package, and would need to be followed by `roxygenize()` to actually get `importFrom(dplyr, mutate)` registered in the NAMESPACE.

**Comment by AshesITR** [comment]
If your `a.R` is a helper script without side effects, you could source it before linting to have the definitions available.

Try
```r
source("a.R")
lintr::lint("b.R")
```

**Comment by bakaburg1** [comment]
Hello! 

same problem here while package development:
<img width="604" alt="image" src="https://github.com/user-attachments/assets/3fae47d3-6a30-4e8a-8f57-09aeb1d70022">

Even if I imported dplyr in the namespace:

```r
# NAMESPACE
import(dplyr)
importFrom(dplyr,bind_rows)
```
Is it a problem of how vscode (or some R extension) running the lintr?

**Comment by AshesITR** [comment]
Potentially. Please try using `lintr::lint_package()` directly to eliminate other possible sources of error. Also don't forget to use `pkgload::load_all()` before linting.
